core/src/main/scala/kafka/server/ControllerApis.scala

```diff
@@ -229,9 +229,9 @@ class ControllerApis(
     // Check if topic deletion is enabled at all.
     if (!config.deleteTopicEnable) {
       if (apiVersion < 3) {
-        throw new InvalidRequestException("Topic deletion is disabled.")
+        return CompletableFuture.failedFuture(new InvalidRequestException("Topic deletion is disabled."))
       } else {
-        throw new TopicDeletionDisabledException()
+        return CompletableFuture.failedFuture(new TopicDeletionDisabledException("Topic deletion is disabled"))
```

Comment:
The `TopicDeletionDisabledException` is pretty self explanatory. Do we need the message `Topic deletion is disabled` in it?